src/util/serialize.h

```diff
@@ -191,6 +191,11 @@ inline f32 readF1000(const u8 *data)
 	return (f32)readS32(data) / FIXEDPOINT_FACTOR;
 }
 
+inline double readD1000(const u8 *data)
+{
+	return (double)readS32(data) / FIXEDPOINT_FACTOR;
```

Comment:
<pre>
> -31728454 / (double)1000
-31728.454

> -31728454 / (float)1000
-31728.453
</pre>
The "x1000" format only ever had a precision of NNNNNNN.NNN, so is the difference between float/double here really significant enough to warrant adding this?

The other question is why would anyone use D1000 over D? To my knowledge remaining F1000 usage is mostly legacy.

Reply:
Yes, the difference is significant. After about 16777.216 (node 168), single precision no longer has enough precision to represent all three digits in the F1000 format. By node 31000, it does not have enough precision to fully represent even one decimal past the fixed point.
I don't have enough experience with developing Minetest to judge how important F1000 format is to preserve. It seems to be limited to some client-side persistence, as far as I could tell.
I can remove the code for it based on your judgment.

Reply:
I removed the D1000 serialization.
The getF1000/putF1000 set of serialization did not include support for F32 serialization. I found substantial use of these functions by various kinds of entity in the server, including for persisting position information. I added missing support for getF32/putF32 and then for the getF64/putF64 serialization.